### PR TITLE
III-4365 Refactor the place import code to PSR15

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added `ImportTermRequestBodyParser` to correctly import terms
- Added `ImportPlaceRequestHandler` to handle the import

### Removed
- Removed `PlaceDocumentImporter`

---
Ticket: https://jira.uitdatabank.be/browse/III-4365 AND https://jira.uitdatabank.be/browse/III-4541

### CODE REVIEW

File: app/Place/PlaceImportServiceProvider.php [hunk 1]
```diff
@@ -35,30 +36,28 @@ function (Application $app) {
...
+                    )
+                ),
+                $app['place_iri_generator'],
+                $app['imports_command_bus'],
+                $app['import_image_collection_factory'],
+                $app['labels.labels_locked_for_import_repository'],
+                $app['should_auto_approve_new_offer'],
+                $app['auth.api_key_reader'],
+                $app['auth.consumer_repository']
+            )
```
Comment (bertramakers) [inline app/Place/PlaceImportServiceProvider.php:60]: Maybe move this request handler definition to `PlaceControllerProvider` with the other place request handlers, and remove this service provider. (Like Organizers)

File: src/Http/Import/ImportTermRequestBodyParser.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if (isset($json->terms) && is_array($json->terms)) {
+            $json->terms = array_map(
+                function (stdClass $term) {
+                    if (isset($term->id) && is_string($term->id)) {
+                        $id = $term->id;
+                        $category = $this->placeCategoryResolver->byId(new CategoryID($id));
+
+                        if ($category) {
+                            $term->{'label'} = $category->getLabel()->toString();
+                            $term->{'domain'} = $category->getDomain()->toString();
```
Comment (bertramakers) [inline src/Http/Import/ImportTermRequestBodyParser.php:36]: I think you can just use `$term->label` and `$term->domain` here, no need for the curly braces and quotation marks.

**Review by bertramakers — APPROVED** [review]
Just some small suggestions, otherwise this looks great!

File: app/Place/PlaceImportServiceProvider.php (hunk 1 above)
Comment (LucWollants) [reply app/Place/PlaceImportServiceProvider.php]: Indeed a cleaner approach.

File: src/Http/Import/ImportTermRequestBodyParser.php (hunk 2 above)
Comment (LucWollants) [reply src/Http/Import/ImportTermRequestBodyParser.php:36]: Indeed the curly brackets are not needed. I had the `$term` variable first typed as `object` when doing that PHPStan gives following errors:
```
 ------ -------------------------------------------------- 
  Line   src/Http/Import/ImportTermRequestBodyParser.php   
 ------ -------------------------------------------------- 
  35     Access to an undefined property object::$label.   
  36     Access to an undefined property object::$domain.  
 ------ -------------------------------------------------- 
 ```

When changing the type to stdClass those errors are gone.

File: src/Http/Place/ImportPlaceRequestHandler.php [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        }
+
+        /** @var Place $place */
+        $place = RequestBodyParserFactory::createBaseParser(
+            new IdPropertyPolyfillRequestBodyParser($this->iriGenerator, $placeId),
+            $this->importPreProcessingRequestBodyParser,
+            new DenormalizingRequestBodyParser($this->placeDenormalizer, Place::class)
+        )->parse($request)->getParsedBody();
+
+        $placeAdapter = new Udb3ModelToLegacyPlaceAdapter($place);
```
[the remark this answers is not in the document]
Comment (LucWollants) [reply src/Http/Place/ImportPlaceRequestHandler.php:126]: Marked the `LegacyOffer` classes and its derived classes as deprecated.

**Pull request merged**